src/main/java/org/opensearch/knn/index/codec/KNN80Codec/KNN80DocValuesConsumer.java

```diff
@@ -112,7 +112,18 @@ private KNNEngine getKNNEngine(@NonNull FieldInfo field) {
     }
 
     private VectorTransfer getVectorTransfer(FieldInfo field) {
-        if (VectorDataType.BINARY.getValue().equalsIgnoreCase(field.attributes().get(KNNConstants.VECTOR_DATA_TYPE_FIELD))) {
```

Comment:
Let's move this check to Binary in differnt functions , It's make easy to understand the flow.

```
boolean isBinary = isFieldBinary(field);

private boolean isFieldBinary(Field field) {
    // Check if the field has a model ID and retrieve the model's vector data type
    if (field.attributes().containsKey(MODEL_ID)) {
        Model model = ModelCache.getInstance().get(field.attributes().get(MODEL_ID));
        return model.getModelMetadata().getVectorDataType() == VectorDataType.BINARY;
    } else if (VectorDataType.BINARY.getValue().equalsIgnoreCase(field.attributes().get(KNNConstants.VECTOR_DATA_TYPE_FIELD))) {
        return true;
    }
    return false;
}
```